zhaquirks/onesti/easycodetouch.py

```diff
@@ -0,0 +1,99 @@
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.closures import DoorLock
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Identify,
+    Ota,
+    PowerConfiguration,
+    Scenes,
+)
+import zigpy.zdo.types
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    NODE_DESCRIPTOR,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+MANUFACTURER_SPECIFIC_PROFILE_ID = 0xFEA2
+
+
+class BatteryBasicCluster(CustomCluster, Basic):
+    """Basic cluster implementation.
+    This implementation sets power source to battery
+    and removes incorrect MainsPowered-flag.
+    """
+
+    cluster_id = Basic.cluster_id
+    POWER_SOURCE = 0x0007
+
+    _CONSTANT_ATTRIBUTES = {
+        POWER_SOURCE: Basic.PowerSource.Battery,
```

Comment:
Just curious, is this "power source" attribute actually used anywhere (in HA)?
Showing the battery percentage sensor only relies on the node descriptor IIRC(?)

Reply:
This was my initial attempt to make it act as a battery-powered device.
I've tested and you're right. Power source attribute is not required. I've removed it.